packages/sdk/src/clients/optimisticOracle/README.md

```diff
@@ -0,0 +1,27 @@
+# UMA SDK Optimistic Oracle Client
+
+This is a basic wraper around the typechain Optimistic Oracle contract which adds some event decoding.
+
+## Usage
+
+Connect to a typechain contract instance and recreate state from events.
+
+```js
+import {ethers} from 'ethers'
+import uma from '@uma/sdk'
+
+// assume you have a url injected from env
+const provider = new ethers.providers.WebSocketProvider(env.CUSTOM_NODE_URL)
+
+// get the contract instance
+const contractAddress:string = // assume you have an lsp address you want to connect to
+const client:uma.clients.optimisticOracle.Instance = uma.clients.optimisticOracle.connect(contractAddress,provider)
+// gets all events using etheres query filter api
+const events = await client.queryFilter({})
+
+// returns EventState, defined in the lsp client
+const state:uma.clients.optimisticOracle.EventState = uma.clients.optimisticOracle.getEventState(events)
```

Comment:
Why not just have the user call your function to query and parse the events? Is it better to have them handle the raw events themselves?

Reply:
i think because i didnt want to assume the source of the events, and i wanted the ability to test without having a live provider. though it is really hard to serialize these events to run later.